game.cpp

```diff
@@ -2,305 +2,308 @@
 
 #include <SDL.h>
 
-#include "r_itexture.h"
+#include <utility>
+
+#include "CWorld.h"
+#include "Shape.h"
+#include "ShapeSphere.h"
 #include "camera.h"
-#include "input.h" 
+#include "input.h"
 #include "physics.h"
+#include "r_itexture.h"
 #include "utils.h"
-#include "Shape.h"
-#include "ShapeSphere.h"
-#include "CWorld.h"
 
 #include "imgui.h"
 
-
 #define MAP_PARSER_IMPLEMENTATION
 #include "map_parser.h"
 
 #include "polysoup.h"
 
 static int hkd_Clamp(int val, int clamp) {
-    if (val > clamp || val < clamp) return clamp;
-    return val;
+	if (val > clamp || val < clamp)
+		return clamp;
+	return val;
 }
 
-Game::Game(std::string exePath, hkdInterface* interface, IRender* renderer)
-{
-    m_Renderer = renderer;
-    m_Interface = interface;
-    m_ExePath = exePath;
+Game::Game(std::string exePath, hkdInterface *interface, IRender *renderer) {
+	m_Renderer = renderer;
+	m_Interface = interface;
+	m_ExePath = std::move(exePath);
```

Comment:
hmm, i don't know about this..

Reply:
I hope we won't litter our code with any of the STL's move semantics.

Reply:
i will try to remove those changes again. i know to little about this yet